subgraphs/level-finance/src/modules/transaction.ts

```diff
@@ -31,11 +32,23 @@ export function transaction(
   const poolInputTokens = pool.getInputTokens();
   const idx = pool.getInputTokens().indexOf(token.id);
   const inputTokenBalances = pool.pool.inputTokenBalances;
-  if (transactionType === TransactionType.DEPOSIT) {
-    inputTokenBalances[idx] = inputTokenBalances[idx].plus(amount);
-  }
-  if (transactionType === TransactionType.WITHDRAW) {
-    inputTokenBalances[idx] = inputTokenBalances[idx].minus(amount);
+  const network = dataSource.network();
+  if (utils.equalsIgnoreCase(network, constants.Network.ARBITRUM_ONE)) {
+    const poolContract = PoolContract.bind(
+      Address.fromBytes(pool.getBytesID())
+    );
+    const tokenBalance = utils.readValue(
+      poolContract.try_poolBalances(Address.fromBytes(token.id)),
+      constants.BIGINT_ZERO
+    );
+    inputTokenBalances[idx] = tokenBalance;
+  } else {
```

Comment:
Is BSC not subject to the same negative balance issues? Or does bsc not have the `try_poolBalances()` function available to call?

Reply:
It does not have the negative balance issue, but I understand your point that it could be there in the future, so should I add the poolBalances function for BSC too? I didn't add it till now because we generally try to avoid network calls.

Reply:
I think the current school of thought is to avoid read calls to the chain if they can be replicated with event data. From my understanding + / - doesn't replicate `try_poolBalances()` exactly

Reply:
Added similar function to bsc too.